hostinfo/packages.lua

```diff
@@ -30,47 +31,100 @@ end
-        })
-      end
-    end
-    callback()
-  end
-
-  local command, args, options, vendor
+  local pkgcmd, pkgargs, options, statcmd, statargs, vendor, method, status, errors, packages, autoupdates
   vendor = sigar:new():sysinfo().vendor:lower()
+  errors, packages, autoupdates = {}, {}, {}
 
   if vendor == 'ubuntu' or vendor == 'debian' then
-    command = 'dpkg-query'
-    args = {'-W'}
+    pkgcmd = 'dpkg-query'
+    statcmd = 'apt-config'
+    pkgargs = {'-W' }
+    statargs = {'dump' }
+    method = 'unattended_upgrades'
     options = {}
   elseif vendor == 'rhel' or vendor == 'centos' then
-    command = 'rpm'
-    args = {"-qa", '--queryformat', '%{NAME}: %{VERSION}-%{RELEASE}\n'}
+    pkgcmd = 'rpm'
+    statcmd = 'service'
+    pkgargs = {"-qa", '--queryformat', '%{NAME}: %{VERSION}-%{RELEASE}\n' }
+    statargs = {'yum-cron', 'status' }
+    method = 'yum_cron'
     options = {}
   else
-    self._error = 'Could not determine OS for Packages'
-    callback()
-    return
+    self._error = 'Could not determine linux distro for Packages'
+    return callback()
   end
 
-  execFileToBuffers(command, args, options, execCb)
+  async.parallel({
+    function(cb)
+    local function pkgExecCb(err, exitcode, stdout_data, stderr_data)
+      if exitcode ~= 0 then
+        table.insert(errors, fmt("Packages exited with a %d exitcode", exitcode))
+        return cb()
+      end
+      for line in stdout_data:gmatch("[^\r\n]+") do
+        line = line:gsub("^%s*(.-)%s*$", "%1")
+        local _, _, key, value = line:find("(.*)%s(.*)")
+        if key ~= nil and value ~= nil and key ~=  '' and value ~= '' then
+          table.insert(packages, {
+            name = key,
+            version = value
+          })
+        end
+      end
+      return cb()
+    end
+    execFileToBuffers(pkgcmd, pkgargs, options, pkgExecCb)
+
+  end,
+    function(cb)
+    local function statExecCb(err, exitcode, stdout_data, stderr_data)
+      status = 'disabled'
+      if exitcode ~= 0 then
+        table.insert(errors, fmt("Autoupdates check exited with a %d exitcode", exitcode))
+        return cb()
+      end
+      if vendor == 'rhel' or vender == 'centos' then
+        status = 'enabled'
+      elseif vendor == 'ubuntu' or vendor == 'debian' then
+        for line in stdout_data:gmatch("[^\r\n]+") do
+          local _, _, key, value = line:find("(.*)%s(.*)")
+          value, _ = value:gsub('"', ''):gsub(';', '')
+          if key == 'APT::Periodic::Unattended-Upgrade' and value ~= 0 then
+            status = 'enabled'
+          end
+        end
+      end
+      return cb()
+    end
+    execFileToBuffers(pkgcmd, pkgargs, options, statExecCb)
+
+  end
+  }, function(err)
+    table.insert(errors, err)
+    if err and not next(autoupdates) and not next(packages) then
+      self._errors = table.concat(errors, ' ')
+      return callback()
+    end
+    if update_method ~= nil and update_method ~= '' and status ~= '' and status ~= nil then
```

Comment:
if update_method and #update_method > 0 and status and #status > 0 
Changes, think i did that in a couple other places as well so fixed those too